client/my-sites/upgrades/checkout-thank-you/purchase-detail.jsx

```diff
@@ -9,47 +9,50 @@ import React from 'react';
  */
 import Button from 'components/button';
 
-const PurchaseDetail = ( {
-	additionalClass,
-	buttonText,
-	description,
-	href,
-	isPlaceholder,
-	target,
-	title
-} ) => {
-	const classes = classNames( 'checkout-thank-you__purchase-detail', additionalClass, {
-		'is-placeholder': isPlaceholder
-	} );
+const PurchaseDetail = React.createClass( {
+	propTypes: {
```

Comment:
I agree with @scruffian - we have other options here that don't involve converting this to a stateful component:

- `renderButton` could be a pure function at the module scope, instead of a method.
- We could assign the button JSX to a `buttonElement` variable if `buttonText` is present.

The latter is probably better. We _do_ want to use the stateful component syntax when we would otherwise be creating a new function on every render (like when we pass a prop to an event handler with bind()), but that is not the case here.